### PR TITLE
joker: init at 0.8.6

### PR DESCRIPTION
**Pull request by andrestylianos**
###### Motivation for this change

Adding Joker, a Clojure interpreter/linter written in go.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/interpreters/joker/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+buildGoPackage rec {
+  name = "joker-${version}";
+  version = "0.8.5";
+  rev = "5318e469184a12ccc011d33d2d32b01d74254765";
+
+  goPackagePath = "github.com/candid82/joker";
+
+  src = fetchFromGitHub {
+    inherit rev;
```
Comment (orivej) [inline pkgs/development/interpreters/joker/default.nix]: Prefer git tag: `rev = "v${version}"`

File: pkgs/development/interpreters/joker/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  dontInstallSrc = true;
+
+  goDeps = ./deps.nix;
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/candid82/joker;
+    description = "A small Clojure interpreter and linter written in Go";
+    license = licenses.epl10;
+    platforms = stdenv.lib.platforms.all;
```
Comment (orivej) [inline pkgs/development/interpreters/joker/default.nix]: Please delete `stdenv.lib` here and below.

**Review by orivej — CHANGES_REQUESTED** [review]

**Comment by andrestylianos** [comment]
I've opened an issue at upstream as v0.8.5 had a problem that was preventing it from building correctly. Now v0.8.6 has been released, builds correctly here and I've updated the PR. Should I update the PR name or commit message to correctly reflect that it's v0.8.6?

**Comment by orivej** [comment]
Sure, could you squash the commits into one with `0.8.6` in it?

**Comment by orivej** [comment]
Additionally I've prevented the installation of generators that were not meant to be installed into `bin`.

**Pull request merged**